### PR TITLE
ipn/ipnlocal: support serving files/directories too

### PR DESCRIPTION
**Pull request by bradfitz**
Updates tailscale/corp#7515

### CODE REVIEW

File: ipn/ipnlocal/serve.go [hunk 1]
```diff
@@ -219,6 +229,73 @@ func (b *LocalBackend) serveWebHandler(w http.ResponseWriter, r *http.Request) {
 	http.Error(w, "empty handler", 500)
 }
 
+func (b *LocalBackend) serveFileOrDirectory(w http.ResponseWriter, r *http.Request, fileOrDir, mountPoint string) {
+	fi, err := os.Stat(fileOrDir)
+	if err != nil {
+		if os.IsExist(err) {
```
Comment (andrew-d) [inline ipn/ipnlocal/serve.go:235]: `errors.Is(err, fs.ErrNotExist)`?

File: ipn/ipnlocal/serve.go [hunk 2]
```diff
@@ -219,6 +229,73 @@ func (b *LocalBackend) serveWebHandler(w http.ResponseWriter, r *http.Request) {
 	http.Error(w, "empty handler", 500)
 }
 
+func (b *LocalBackend) serveFileOrDirectory(w http.ResponseWriter, r *http.Request, fileOrDir, mountPoint string) {
+	fi, err := os.Stat(fileOrDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			http.NotFound(w, r)
```
Comment (shayne) [inline ipn/ipnlocal/serve.go:236]: missing return?

Comment (shayne) [inline ipn/ipnlocal/serve.go:236]: missing return?

**Review by shayne — CHANGES_REQUESTED** [review]

File: ipn/ipnlocal/serve.go (hunk 2 above)
Comment (bradfitz) [reply ipn/ipnlocal/serve.go:236]: Good catch. Happens to be accidentally harmless in this case, but wrong nonetheless.

Comment (bradfitz) [reply ipn/ipnlocal/serve.go:236]: PTAL

**Review by shayne — APPROVED** [review]

**Pull request merged**